static/js/popover_menus.js

```diff
@@ -87,4 +107,196 @@ export function initialize() {
+                should_display_reminder_option: feature_flags.reminders_in_message_action_menu,
+                should_display_edit_and_view_source,
+                should_display_quote_and_reply,
+            };
+
+            instance.setContent(render_actions_popover_content(args));
+            message_actions_popover_visible = true;
+
+            const $popper = $(instance.popper);
+
+            $popper.one("click", instance.hide);
+
+            $popper.one("click", ".rehide_muted_user_message", (e) => {
+                if (row && !message_container.is_hidden) {
+                    message_lists.current.view.hide_revealed_message(message_id);
+                }
+                e.preventDefault();
+            });
+
+            $popper.one("click", ".popover_edit_message", (e) => {
+                message_edit.start(row);
+                e.preventDefault();
+            });
+
+            $popper.one("click", ".respond_button", (e) => {
+                // Arguably, we should fetch the message ID to respond to from
+                // e.target, but that should always be the current selected
+                // message in the current message list (and
+                // compose_actions.respond_to_message doesn't take a message
+                // argument).
+                compose_actions.quote_and_reply({trigger: "popover respond"});
+                e.preventDefault();
+            });
+
+            $popper.one("click", ".reminder_button", (e) => {
+                popovers.render_actions_remind_popover(
+                    $(".selected_message .actions_hover")[0],
+                    message_id,
+                );
+                e.preventDefault();
+            });
+
+            $popper.one("click", ".popover_toggle_collapse", (e) => {
+                if (row) {
+                    if (message.collapsed) {
+                        condense.uncollapse(row);
+                    } else {
+                        condense.collapse(row);
+                    }
+                }
+                e.preventDefault();
+            });
+
+            $popper.one("click", ".view_edit_history", (e) => {
+                const message_history_cancel_btn = $("#message-history-cancel");
+
+                message_edit_history.show_history(message);
+                message_history_cancel_btn.trigger("focus");
+                e.preventDefault();
+            });
+
+            $popper.one("click", ".delete_message", (e) => {
+                message_edit.delete_message(message_id);
+                e.preventDefault();
+            });
+
+            $popper.one("click", ".popover_mute_topic", (e) => {
+                muting_ui.mute_topic(message.stream_id, message.topic);
+                e.preventDefault();
+            });
+
+            $popper.one("click", ".popover_unmute_topic", (e) => {
+                muting_ui.unmute_topic(message.stream_id, message.topic);
+                e.preventDefault();
+            });
```

Comment:
It's kinda inconvenient for review that the sort order for these was changed in this commit, since it makes `git show` do a much less job of showing moved code; next time, if you're going to re-sort, do that in a prep commit.